### PR TITLE
ci: Allow manual execution for deploy argilla environment

### PR DESCRIPTION
**Pull request by frascuchon**
This PR configs the manual execution of the PR environments.

Also, clean running environments once per day.

### CODE REVIEW

**Review by gabrielmbmb — APPROVED** [review]

File: .github/workflows/deploy-environment.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
 name: Deploy Argilla environment
 
 on:
+  workflow_dispatch:
```
Comment (jfcalvo) [inline .github/workflows/deploy-environment.yml:4]: So with this you can click on run and manually create a pr environment for the PR on gcloud?

File: .github/workflows/deploy-environment.yml [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
   workflow_call:
     inputs:
       image-name:
         description: The name of the Docker image to deploy.
         type: string
+        default: argilla/argilla-quickstart-for-dev
```
Comment (jfcalvo) [inline .github/workflows/deploy-environment.yml:10]: Why did you change this? It's an image that still doesn't exists?

Comment (frascuchon) [reply .github/workflows/deploy-environment.yml:10]: I added the image name as default. This will simplify the manual deployment execution. This image is created in the [previous step](https://github.com/argilla-io/argilla/blob/ci/manual-pr-deploy-and-cleaning-services-one-by-day/.github/workflows/package.yml#L155). At this point, it will be available to use.

File: .github/workflows/deploy-environment.yml (hunk 1 above)
Comment (frascuchon) [reply .github/workflows/deploy-environment.yml:4]: Exactly.

**Review by jfcalvo — APPROVED** [review]
